### DISCUSSION TITLE
C:\ references in the ngrok webpage

### DISCUSSION THREAD

**Discussion by amalcolm** [kinngh/shopify-node-express-mongodb-app#47] [Q&A]
Thank you for doing such a splendid template and video for Shopify apps.
I've managed to get 99% there but can't get the last 1%.

**I'm getting C:\ references in the ngrok webpage for my app.**  I.e. that opened with the .ngrok.io domain.  This results in the app failing to install to the shop.

I believe the cause is I'm not specifying the host somewhere, but I've no idea where.

This belief comes a as in client.js 
```
 if (!config.host) {
        throw Error_1.fromAction('host must be provided', Error_1.AppActionType.INVALID_CONFIG);
    }
```
is throwing  The config is generated by a worker and I was unable to trace back any further (react-dom.development.js).

The outcome is that C:\ is passed to the website, for instance, entry-client.jsx:

```
import { jsxDEV } from "/@id/__x00__react/jsx-dev-runtime";
import __vite__cjsImport1_reactDom_client from "/@fs/C:/Users/.../node_modules/.vite/deps/react-dom_client.js?v=fd5fc421"; 
...
root.render(...
  fileName: "C:/Users/.../client/entry-client.jsx",
  ...

}, this));
```

Yes, it's Windows, and I had to change package.json to set NOTE_ENV=dev && ... which seemed to work.  
Had to update Node.js to avoid the node:path error (for Node.js < 16 evidently).

I've filled in the .env with seemingly valid stuff, have a custom domain for ngrok in the scripts/ngrok
 other than that everything is from the template.

It's been a long day and I'm probably overlooking something super simple.  

Thanks in advance for any insights.

**Comment by kinngh** [comment]
Since you're on Windows, can you try replacing all instances of reading to and from a directory with a `path` instead? As an example, in `server/index.js:157`, 

`.send(fs.readFileSync(${root}/dist/client/index.html));` 
becomes 
`.send(fs.readFileSync(path.join(root, "dist", "client", "index.html")));`



If this doesn't work out, please ping me here or over on [Twitter](https://twitter.com/kinngh) and we could possibly hop on a call!
